UPGRADING-HADOOP.md

```diff
@@ -0,0 +1,181 @@
+# Hadoop Upgrade
+## Actions and Instructions
+
+The hadoop-base layer contains an action called hadoop-upgrade which allows the
+administrator to upgrade, downgrade and finalize a hadoop deployment. Currently
+rollback is not supported, and HDFS filesystem version upgrade is not supported (but
+will be implemented when a new filesystem version is released greater than the version
+which comes with 2.7.2)
+
+The actions and the commands they run are in line with the instructions and
+recommendations in this document:
+
+http://hadoop.apache.org/docs/r2.7.2/hadoop-project-dist/hadoop-HDFS/HDFSRollingUpgrade.html
+
+Therefore, when upgrading a namenode, datanode or resourcemanager, the following
+daemons will be stopped/started: 
+  namenode  
+  datanode
+  nodemanager
+  resourcemanager
+  jobhistoryserver
+
+While ALL hadoop binaries are upgraded by this process, neither journalnode nor
+zkfc (zookeeper) daemons are restarted by the upgrade process, as doing so can
+cause an HDFS outage. Further instructions for restarting these daemons after an
+upgrade are provided in the action definitions in: the datanode layer (for
+journalnode) and the namenode layer for zkfc. 
+
+  
+### Upgrade (HA)
+
+The apache documentation suggests the following process for performing a rolling
+upgrade - that is, an HDFS upgrade without downtime:
+
+  1. Prepare rolling upgrade
+  2. Upgrade namenodes
+  3. Upgrade (a subset of) datanodes
+  4. Finalize (or downgrade, or rollback*) 
+
+There is an additional 'safety' step with our hadoop implementations: you must
+config set the version that you would like to upgrade (or downgrade) to for each
+service you are going to upgrade. 
+
+Here are the general steps:
+
+1. Set hadoop_version config value for each service:
+
+
+    version=2.7.2
+    juju set namenode hadoop_version=$version
+    juju set slave hadoop_version=$version
+2. Prepare the rolling upgrade:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version prepare=true
+3. Upgrade both namenodes:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version
```

Comment:
Wait up, you are correct - however, if this user copy/pastes this, the first namenode will be upgraded before the second is shut down. Perhaps you could make this an addendum to your pull request.